src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLiteConnectionPoolService.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.SQLite.v2.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SQLite.v2
+{
+    [Export]
+    [Shared]
+    internal sealed class SQLiteConnectionPoolService : IDisposable
+    {
+        private const string LockFile = "db.lock";
+
+        /// <summary>
+        /// Maps from database path to connection pool.
+        /// </summary>
+        private readonly Dictionary<string, ReferenceCountedDisposable<SQLiteConnectionPool>> _connectionPools = new();
```

Comment:
might want to mention what is used to loc this (presumably itself).

Reply:
➡️ Now documented